src/lib/components/form/Form.component.tsx

```diff
@@ -0,0 +1,346 @@
+        alignItems={'baseline'}
+        gap={direction === 'horizontal' ? spacing['r32'] : spacing['r4']}
+      >
+        <div
+          style={{
+            width: `${
+              maxLabelWidth === 0 ? 'max-content' : `${maxLabelWidth}px`
+            }`,
+            opacity: disabled ? 0.5 : 1,
+          }}
+        >
+          <Stack>
+            <label htmlFor={id} ref={labelRef}>
+              <Text>
+                {label}
+                {requireMode !== 'all' && required && ' *'}
+                {requireMode === 'all' && !required && ' (optional)'}
+              </Text>
+            </label>
+            {labelHelpTooltip && (
+              <IconHelp
+                tooltipMessage={labelHelpTooltip}
+                overlayStyle={{ maxWidth: '20rem' }}
+              />
+            )}
+          </Stack>
+        </div>
+        <Stack
+          direction={helpErrorPosition === 'right' ? 'horizontal' : 'vertical'}
+          gap={helpErrorPosition === 'right' ? 'r8' : 'r4'}
+        >
+          {content}
+          {error ? (
+            <Text
+              variant="Smaller"
+              color="statusCritical"
+              isEmphazed
+              id={`${DESCRIPTION_PREFIX}${id}`}
+            >
+              {error}
+            </Text>
+          ) : help ? (
+            <div
+              style={{
+                opacity: disabled ? 0.5 : 1,
+              }}
+            >
+              <Text
+                variant="Smaller"
+                color="textSecondary"
+                isEmphazed
+                id={`${DESCRIPTION_PREFIX}${id}`}
+              >
+                {help}
+              </Text>
+            </div>
+          ) : (
+            <Text variant="Smaller" isEmphazed={true}>
+              &nbsp;
+            </Text>
+          )}
+        </Stack>
+      </Box>
+    </FieldContext.Provider>
+  );
+};
+
+type FormSectionProps = {
+  children: ReactElement<FormGroupProps> | ReactElement<FormGroupProps>[];
+  title?: string;
+  icon?: IconName;
+  helpTooltip?: string;
+};
+
+const FormSection = ({
+  children,
+  title,
+  icon,
+  helpTooltip,
+}: FormSectionProps) => {
+  const [maxLabelWidth, setMaxLabelWidth] = useState(0);
+  //If all the formgroup are not required, add `(optional)` next to form section title.
+  const groupNotOptional = Children.toArray(children).find(
+    (child: ReactElement<FormGroupProps>) => child.props.required === true,
+  );
+
+  return (
+    <LabelContext.Provider value={{ maxLabelWidth, setMaxLabelWidth }}>
+      <Stack direction="vertical" gap="r12">
+        {title && (
+          <Stack direction="horizontal" gap="r8">
+            {icon && <Icon name={icon}></Icon>}
+            <Text isEmphazed>
+              {groupNotOptional ? `${title}` : `${title} (optional)`}
+            </Text>
+            {helpTooltip && (
+              <IconHelp
+                tooltipMessage={helpTooltip}
+                overlayStyle={{ maxWidth: '20rem' }}
```

Comment:
Maybe we should extract this `overlayStyle ` in a const common within this file (because we have 2 other tooltip using the same)